launcher/launch.py

```diff
@@ -36,6 +38,85 @@ def join(self):
 COMMON_REQUIRED_ENVS = ["DMLC_ROLE", "DMLC_NUM_WORKER", "DMLC_NUM_SERVER",
                         "DMLC_PS_ROOT_URI", "DMLC_PS_ROOT_PORT"]
 WORKER_REQUIRED_ENVS = ["DMLC_WORKER_ID"]
+NUMA_PATH = "/sys/devices/system/node"
+
+
+def get_numa_info():
+    ret = []
+    if os.path.exists(NUMA_PATH):
+        items = os.listdir(NUMA_PATH)
+        nodes = list(filter(lambda str: str.startswith("node"), items))
+        if nodes:
+            for node in nodes:
+                items = os.listdir(os.path.join(NUMA_PATH, node))
+                cpus = [re.findall("cpu\d+", cpu) for cpu in items]
+                cpus = list(filter(lambda x: x, cpus))
+                cpu_ids = [int(cpu[0].split('cpu')[1]) for cpu in cpus]
+                cpu_ids = sorted(cpu_ids)
+                ret.append(cpu_ids)
+    else:
+        print("NUMA PATH %s NOT FOUND" % NUMA_PATH)
+    return ret
+
+
+def allocate_cpu(local_size):
+    def _get_allocation(nodes, quota):
+        if quota < 1:
+            raise ValueError("quota should be no less than 1")
+        ret = []
+        for node in nodes:
+            if len(node) < quota:
+                continue
+            split_index = []
+            for i in range(1, quota):
+                if node[i] != node[i-1] + 1:
+                    split_index.append(i)
+            quota_bck = quota
+            last_idx = 0
+            for idx in split_index:
+                ret.append(node[last_idx:idx])
+                quota -= idx - last_idx
+                last_idx = idx
+            ret.append(node[last_idx:last_idx+quota])
+            for idx in sorted(range(quota_bck), reverse=True):
+                del node[idx]
+            return ret
+        return ret
+
+    def _get_quota(nodes, local_size):
+        if len(nodes) > 1:
+            cpu_nums = reduce(lambda x, y: (len(x) + len(y)), nodes)
+        else:
+            cpu_nums = len(nodes[0])
+
+        default_quota = int(os.getenv("BYTEPS_NUMA_DEFAULT_QUOTA", 6))
+        while default_quota >= 1 and default_quota * local_size > cpu_nums:
+            default_quota -= 2
+
+        root_quota = cpu_nums - default_quota * (local_size - 1)
+        if int(os.getenv("BYTEPS_NUMA_ROOT_QUOTA", 0)):
```

Comment:
Can you add some comments here to explain the usage of `BYTEPS_NUMA_DEFAULT_QUOTA`  and `BYTEPS_NUMA_ROOT_QUOTA`?